pkg/controller/nodekeeper/nodekeeper_controller.go

```diff
@@ -134,9 +136,23 @@ func (r *ReconcileNodeKeeper) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
-	isNodeDrainTimedOut := result.StartTime != nil && result.StartTime.Add(cfg.NodeDrain.GetDuration()).Before(metav1.Now().Time)
+	drainStrategy, err := r.drainstrategyBuilder.NewNodeDrainStrategy(r.client, uc, node, &cfg.NodeDrain)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	res, err := drainStrategy.Execute(result.StartTime)
+	for _, r := range res {
+		reqLogger.Info(r.Message)
+	}
+	if err != nil {
```

Comment:
The return result is an array of strategy results:
```go
Execute(*metav1.Time) ([]*DrainStrategyResult, error)
```
We can have some successful drain results but also errors. It will never be `nil`

That ok do you think?

Reply:
Ah okay, that makes sense in context - cheers.